common/client-sdk/grpcweb_client.go

```diff
@@ -0,0 +1,90 @@
+//go:build js && wasm
+// +build js,wasm
+
+package arkclient
+
+import (
+	"syscall/js"
+
+	arkv1 "github.com/ark-network/ark/api-spec/protobuf/gen/ark/v1"
+	"google.golang.org/grpc"
+)
+
+func NewGrpcWebClient(aspUrl string) (ArkGrpcWebClient, func(), error) {
+	if aspUrl == "" {
+		return nil, nil, errAspUrlEmpty
+	}
+
+	creds := insecure.NewCredentials()
+	port := 80
+	if strings.HasPrefix(aspUrl, "https://") {
+		aspUrl = strings.TrimPrefix(aspUrl, "https://")
+		creds = credentials.NewTLS(nil)
+		port = 443
+	}
+	if !strings.Contains(aspUrl, ":") {
+		aspUrl = fmt.Sprintf("%s:%d", aspUrl, port)
+	}
+	conn, err := grpc.NewClient(aspUrl, grpc.WithTransportCredentials(creds))
```

Comment:
is this really using a gRPC web client? Seems to be same as gRPC native?

Reply:
> From what I understood, when compiling to WASM targets, we should use gRPC-web like the example I found. Not sure if using JSON interface instead, we can then do code-gen of client stubs like with gRPC and gRPC web?

Wasm [file](https://github.com/johanbrandhorst/grpcweb-wasm-example/blob/master/frontend/html/test.wasm) is generated using [this](https://github.com/johanbrandhorst/grpcweb-wasm-example/blob/50cb9e39fa989ec5c2af9d3200e2d278cf72c034/Makefile#L5) cmd, if u check that [file](https://github.com/johanbrandhorst/grpcweb-wasm-example/blob/master/frontend/frontend.go) u will see in main it creates usual grpc connection. If you check [server](https://github.com/johanbrandhorst/grpcweb-wasm-example/blob/50cb9e39fa989ec5c2af9d3200e2d278cf72c034/main.go#L40) which is hosting FE files(wasm, html) and exposing grpc API's you will see that it is using [grpcweb](https://github.com/johanbrandhorst/grpcweb-wasm-example/blob/50cb9e39fa989ec5c2af9d3200e2d278cf72c034/main.go#L40).